base/floatfuncs.jl

```diff
@@ -273,8 +272,10 @@ julia> isapprox(1e-10, 0, atol=1e-8)
 true
 ```
 """
-function isapprox(x::Number, y::Number; atol::Real=0, rtol::Real=rtoldefault(x,y,atol), nans::Bool=false)
-    x == y || (isfinite(x) && isfinite(y) && abs(x-y) <= max(atol, rtol*max(abs(x), abs(y)))) || (nans && isnan(x) && isnan(y))
+function isapprox(x::Number, y::Number;
+                  atol::Real=0, rtol::Real=rtoldefault(x,y,atol),
+                  nans::Bool=false, norm::Function=abs)
```

Comment:
I don't think the restriction to `Function` is necessary here.

Reply:
It's `norm::Function` in the LinearAlgebra definition, so if the signature is relaxed here it should be relaxed there as well.

Reply:
I suspect that this is one of those cases where `::Function` is fine; it's hard to imagine anyone needing some other functor argument here, and declaring it as `Function` is probably clearer?

Reply:
If we change this in `Base`, we should also change it in `LinearAlgebra` for consistency. However, this would fundamentally change the nature of this PR: Instead of *adding new features*, it would *modify existing behavior*. While the modification is arguable not going to break any existing code, I still think it's an unnecessary complication and unnecessarily complicates this relatively straightforward PR.

I also agree with @stevengj in that restriction to `::Function` this is not such a big deal in this case. However, if there is a strong consensus that `::Function` should die, I propose that it should be dealt with in a new issue.

Reply:
Ok, we can just leave it for now and this can always be a follow-up.